aicssegmentation/workflow/structure_wrapper_config.py

```diff
@@ -0,0 +1,146 @@
+        """
+        Decode Functions config (all_functions.json)
+        """    
+        def build_function_parameter(name: str, data: Dict):
+            return FunctionParameter(name=name,
+                                     widget_type=WidgetType.from_str(data["widget_type"]),
+                                     data_type=data["data_type"],
+                                     min_value=data.get("min", None),
+                                     max_value=data.get("max", None),
+                                     increment=data.get("increment", None),
+                                     options=data.get("options", None)
+                                    )
+        
+        functions = list()
+        for function_k, function_v in obj.items():
+            function = SegmenterFunction(name = function_k,
+                                         display_name=function_v["name"],
+                                         function=function_v["python::function"],
+                                         module=function_v["python::module"])
+
+            if(function_v.get("parameters") is not None and len(function_v["parameters"]) > 0):
+                params = dict()
+
+                for param_k, param_v in function_v["parameters"].items():
+                    param_name = param_k
+                    params[param_name] = list()
+                    
+                    if isinstance(param_v, dict):  
+                        params[param_name].append(build_function_parameter(param_name, param_v))                        
+                    elif isinstance(param_v, list):
+                        for item in param_v:                                                   
+                            params[param_name].append(build_function_parameter(param_name, item) )
+                
+                function.parameters = params
+
+            functions.append(function)
+
+        return functions
+            
+
+    @classmethod
+    def _workflow_decoder(cls, obj: Dict, workflow_name: str) -> WorkflowDefinition:
+        """
+        Decode Workflow config (conf_{workflow_name}.json)
+        """
+        functions = cls.get_all_functions()
+        steps = list()
+
+        for step_k, step_v in obj.items():
+            step_number = int(step_k)
+            function = next(filter(lambda f: f.name == step_v["function"], functions))
+            
+            if isinstance(step_v["parent"], list): #
+                parent = step_v["parent"]
+            else:
+                parent = [step_v["parent"]]
+
+
+            step = WorkflowStep(category=WorkflowStepCategory.from_str(step_v["category"]),
+                                function=function, 
+                                step_number=step_number, 
+                                parent=parent)
+
+            if(step_v.get("parameter_defaults") is not None and len(step_v["parameter_defaults"]) > 0):
```

Comment:
Just to make sure I understand this line:

if this statement if False, then it just means this is a function without parameters, and certaintly not an error to catch. Is this correct?

Reply:
Correct. The code expects to find functions without parameters and/or workflow confs that don't define any parameter defaults. Those should still be loaded